examples/go.mod

```diff
@@ -0,0 +1,139 @@
+module github.com/ipfs/go-libipfs/examples
+
+go 1.19
+
+require (
+	github.com/ipfs/go-blockservice v0.5.0
+	github.com/ipfs/go-cid v0.3.2
+	github.com/ipfs/go-datastore v0.6.0
+	github.com/ipfs/go-ipfs-blockstore v1.2.0
+	github.com/ipfs/go-ipfs-chunker v0.0.5
+	github.com/ipfs/go-ipfs-exchange-offline v0.3.0
+	github.com/ipfs/go-libipfs v0.4.0
+	github.com/ipfs/go-merkledag v0.9.0
+	github.com/ipfs/go-unixfs v0.4.2
+	github.com/libp2p/go-libp2p v0.24.2
+	github.com/libp2p/go-libp2p-routing-helpers v0.6.0
+	github.com/multiformats/go-multiaddr v0.8.0
+	github.com/multiformats/go-multicodec v0.7.0
```

Comment:
I think the only dep that isn't in `github.com/ipfs/go-libipfs`'s `go.mod` is `github.com/libp2p/go-libp2p-routing-helpers v0.6.0`.

I think you should just keep one `go.mod` file, it gets annoying to update multiple `go.mod` files, either you forget and get reminded by CI, or you don't and they get out of sync.

Reply:
My question is whether that will scale to other examples we may do after?

Reply:
@aschmahmann since we've merged two examples in the meanwhile, would you mind rebasing this on the current code so I can have an idea of the changes you're suggesting?

> I'd like examples to also always be usable from only released repos if possible.

I think this would require us to only update examples after releasing, which can lead us to have out of date examples that no longer work.

Reply:
Yep, will rebase.

> I think this would require us to only update examples after releasing, which can lead us to have out of date examples that no longer work.

Well they'd always work, they just wouldn't work with the newest library. Ideally when there are breaking changes there's some deprecation period where we encourage people to switch over. To some extent watching all of our examples break/require hard-coding to a go-libipfs commit (or to `..`) allows us to feel some of that pain and consider what this will mean for our downstream users.

However, the key there was "if possible", sometimes its not or the tradeoff isn't worthwhile. That being said if people prefer this route that's fine as long as we add CI checks to make sure the dependencies get updated (i.e. if we're going the `..` method I don't want go-libipfs depending on go-multihash v0.3.0 while the example depends on v0.2.0 which is implicitly being upgraded.

Reply:
> as we add CI checks to make sure the dependencies get updated

The CI currently already checks if all `go.mod`s are updated, so I think we're good on that.